### ISSUE TITLE
Field error does not display when using AppendText

### ISSUE THREAD

**Issue by dedayoa** [django-crispy-forms/django-crispy-forms#753]
* Package version: 1.7.0
* Django version: 1.11.4
* Python version: 3.6.1
* Template pack: (Optional) Bootstrap4

### Description:
The helper portion of my form is as below...
```
self.helper = FormHelper()
self.helper.form_action = '.'
self.helper.form_tag = False
self.helper.layout = Layout(
    AppendedText('username', "@%s"%(self.domain), active=True),
    Field('email', type="email"),
    Field('first_name'),
    Field('last_name'),
)
```
When the username field fails validation (and an error is generated), it does not display. In devtools, I see that the style to display the error message does not apply because it has been moved out of being a sibling.

```
<div class=""> 
  <div class="input-group"> 
  <input type="text" name="username" value="sdgrgksjog+ff" class="textinput textInput form-control is-invalid" required="" id="id_username">

    <span class="input-group-addon active">@tekartng.com</span> 
  </div>
<span id="error_1_id_username" class="invalid-feedback"><strong>Enter a valid username. This value may contain only letters and numbers.</strong></span> 
 </div>
```

Moving the error message around the input field does not help.

**Comment by carltongibson** [comment]
So `span#error_1_id_username` needs to be inside the `div.input-group`?

BS4 support is a little stalled at the moment. If you could make a PR correcting this it would be great!

**Comment by dedayoa** [comment]
I see this issue has been reported in the bootstrap issues. https://github.com/twbs/bootstrap/issues/23454

**Comment by dedayoa** [comment]
Since this is a pending issue on BS4, the below is what I'm using...
`
.input-group ~ .invalid-feedback{
	display: block;
}`

**Comment by carltongibson** [comment]
OK. Good follow-up. Thanks. 

In general I think the BS4 support is close but the are still issues. 

If you’re using it, any input you can provide to clean up remaining issues would be very welcome! 

I’m going to close this here now. If there’s some things we can actually address I’m happy to reopen.

**Comment by dedayoa** [comment]
Alright Carlton. That's fine.

**Comment by santiagorevale** [comment]
Hi there,

I've tried building the form using `{% crispy form %}` and the same is still happening (field error not being displayed because span is outside the input-group div).

I'm using bootstrap v4.3.1 and django-crispy-forms v1.7.2.

Any progress on either side on how to fix this?

> Since this is a pending issue on BS4, the below is what I'm using...
.input-group ~ .invalid-feedback{ display: block; }

@dedayoa do you mind explaining exactly how to implement the temporary fix you mentioned above? My CSS skills are quite bad.

Thanks in advance.

**Comment by carltongibson** [comment]
Reopening to have another look.

**Comment by csl619** [comment]
Hi Carlton,

Not sure if you have managed to look into this but I think i have found the issue. The `{% include 'bootstrap4/layout/help_text_and_errors.html' %}` is outside of the input group div block on the prepended_appended_text template, if you move that inside of the block the field errors display as intended.

Chris

**Comment by smithdc1** [comment]
Closed by #898?

**Comment by carltongibson** [comment]
Yes, let's assume so. Happy to look at another issue if it comes up in a different context.
